src/lib/hooks/swap/useSwapInfo.tsx

```diff
@@ -1,143 +1,98 @@
+  impact: PriceImpact
 }
 
 // from the current swap inputs, compute the best trade and return it.
 function useComputeSwapInfo(): SwapInfo {
   const { account } = useActiveWeb3React()
   const { type: wrapType } = useWrapCallback()
   const isWrapping = wrapType === WrapType.WRAP || wrapType === WrapType.UNWRAP
-  const {
-    independentField,
-    amount,
-    [Field.INPUT]: inputCurrency,
-    [Field.OUTPUT]: outputCurrency,
-  } = useAtomValue(swapAtom)
+  const { independentField, amount, [Field.INPUT]: currencyIn, [Field.OUTPUT]: currencyOut } = useAtomValue(swapAtom)
   const isExactIn = independentField === Field.INPUT
-  const feeOptions = useAtomValue(feeOptionsAtom)
 
   const parsedAmount = useMemo(
-    () => tryParseCurrencyAmount(amount, (isExactIn ? inputCurrency : outputCurrency) ?? undefined),
-    [inputCurrency, isExactIn, outputCurrency, amount]
+    () => tryParseCurrencyAmount(amount, (isExactIn ? currencyIn : currencyOut) ?? undefined),
+    [amount, isExactIn, currencyIn, currencyOut]
   )
-  // TODO(ianlapham): this would eventually be replaced with routing api logic.
   const trade = useBestTrade(
     isExactIn ? TradeType.EXACT_INPUT : TradeType.EXACT_OUTPUT,
     parsedAmount,
-    (isExactIn ? outputCurrency : inputCurrency) ?? undefined
-  )
-  const tradeCurrencyAmounts = useMemo(
-    () => ({
-      // Use same amount for input and output if user is wrapping.
-      [Field.INPUT]: isWrapping || isExactIn ? parsedAmount : trade.trade?.inputAmount,
-      [Field.OUTPUT]: isWrapping || !isExactIn ? parsedAmount : trade.trade?.outputAmount,
-    }),
-    [isExactIn, isWrapping, parsedAmount, trade.trade?.inputAmount, trade.trade?.outputAmount]
+    (isExactIn ? currencyOut : currencyIn) ?? undefined
   )
-  const slippage = useSlippage(trade.trade)
 
-  const currencies = useMemo(
-    () => ({ [Field.INPUT]: inputCurrency, [Field.OUTPUT]: outputCurrency }),
-    [inputCurrency, outputCurrency]
+  const amountIn = useMemo(
+    () => (isWrapping || isExactIn ? parsedAmount : trade.trade?.inputAmount),
+    [isExactIn, isWrapping, parsedAmount, trade.trade?.inputAmount]
   )
-  const [inputCurrencyBalance, outputCurrencyBalance] = useCurrencyBalances(
-    account,
-    useMemo(() => [inputCurrency, outputCurrency], [inputCurrency, outputCurrency])
+  const amountOut = useMemo(
+    () => (isWrapping || !isExactIn ? parsedAmount : trade.trade?.outputAmount),
+    [isExactIn, isWrapping, parsedAmount, trade.trade?.outputAmount]
   )
-  const currencyBalances = useMemo(
-    () => ({
-      [Field.INPUT]: inputCurrencyBalance,
-      [Field.OUTPUT]: outputCurrencyBalance,
-    }),
-    [inputCurrencyBalance, outputCurrencyBalance]
+  const [balanceIn, balanceOut] = useCurrencyBalances(
+    account,
+    useMemo(() => [currencyIn, currencyOut], [currencyIn, currencyOut])
   )
 
-  const inputError = useMemo(() => {
-    let inputError: ReactNode | undefined
```

Comment:
how are we doing errors here now?

Reply:
These errors weren't actually used anywhere, everything is recomputed in the appropriate child components.
This is cruft from porting from interface.

Reply:
Ahhhhhhh very nice, good catch